common/attest_plugin.c

```diff
@@ -469,47 +491,11 @@ oe_result_t oe_verify_attestation_certificate_with_evidence(
     // find the extension
     OE_TRACE_VERBOSE("extract_x509_report_extension() succeeded");
 
-    if (oid_array_index >= 2) // oid_oe_evidence or oid_new_oe_evidence
```

Comment:
This check is still needed even though you can use the same function/parameters to verify it.

```
If (oid_array_index >= 2)  // evidence format
    it's header version must be OE_ATTESTATION_HEADER_VERSION.
else
    it's header version must be OE_REPORT_HEADER_VERSION

```